### PR TITLE
[LOOP-451] [LOOP 1318] Correction range (& correction range overrides) guardrails

### PR DESCRIPTION
**Pull request by mpangburn**
Requires https://github.com/tidepool-org/LoopKit/pull/101

Implements https://tidepool.atlassian.net/browse/LOOP-451 and (preliminarily) https://tidepool.atlassian.net/browse/LOOP-1318

Pending items to support completion of LOOP-1318:
- The recommend ranges for Pre-Meal and Workout need to be formally defined by Product; this implementation uses 80-120 mg/dL and 100-160 mg/dL, respectively, for the time being.
- The title of the correction range overrides page and the descriptions of the Pre-Meal and Workout correction ranges need formal copy (chatting with Melissa about this).

### CODE REVIEW

File: Loop/View Controllers/SettingsTableViewController.swift [hunk 1]
```diff
@@ -464,29 +473,50 @@ final class SettingsTableViewController: UITableViewController {
...
+                        dataManager?.loopManager.settings.preMealTargetRange = overrides.preMeal?.doubleRange(for: unit)
+                        dataManager?.loopManager.settings.legacyWorkoutTargetRange = overrides.workout?.doubleRange(for: unit)
+                    }
+                )
+
+                let hostingController = ExplicitlyDismissibleModal(rootView: editor, onDisappear: {
+                    tableView.deselectRow(at: indexPath, animated: true)
+                })
+
+                self.present(hostingController, animated: true)
```
Comment (nhamming) [inline Loop/View Controllers/SettingsTableViewController.swift:513]: [nit] is `self` needed? See line 491

File: Loop/View Controllers/SettingsTableViewController.swift [hunk 2]
```diff
@@ -464,29 +473,50 @@ final class SettingsTableViewController: UITableViewController {
...
             case .suspendThreshold:
                 func presentSuspendThresholdEditor(initialValue: HKQuantity?, unit: HKUnit) {
                     let editor = SuspendThresholdEditor(
                         value: initialValue,
                         unit: unit,
+                        maxValue: dataManager.loopManager.settings.glucoseTargetRangeSchedule?.minLowerBound(),
```
Comment (nhamming) [inline Loop/View Controllers/SettingsTableViewController.swift:519]: This seems strange. `maxValue` is set to the `minLowerBound`?

**Review by nhamming — APPROVED** [review]
One item looks questionable, but if you are fine with it, so am I. Otherwise LGTM

**Review by ps2 — APPROVED** [review]
LGTM! 🌶️

File: Loop/View Controllers/SettingsTableViewController.swift (hunk 1 above)
Comment (mpangburn) [reply Loop/View Controllers/SettingsTableViewController.swift:513]: Not needed, will remove in the next PR to this file (up soon)

File: Loop/View Controllers/SettingsTableViewController.swift (hunk 2 above)
Comment (mpangburn) [reply Loop/View Controllers/SettingsTableViewController.swift:519]: By design; it doesn't make sense for the quantity below which Loop suspends all insulin delivery to be able to be higher than user's intended glucose target at any point in time

**Pull request merged**